injectproxy/enforce.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package injectproxy
+
+import (
+	"fmt"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/promql/parser"
+)
+
+type Enforcer struct {
+	labelMatchers map[string]*labels.Matcher
+}
+
+func NewEnforcer(ms ...*labels.Matcher) *Enforcer {
+	entries := make(map[string]*labels.Matcher)
+
+	for _, matcher := range ms {
+		entries[matcher.Name] = matcher
+	}
+
+	return &Enforcer{
+		labelMatchers: entries,
+	}
+}
+
+// EnforceNode walks the given node recursively
+// and enforces the given label enforcer on it.
+//
+// Whenever a parser.MatrixSelector or parser.VectorSelector AST node is found,
+// their label enforcer are being potentially modified.
+//
+// If a node label matcher equals the name with one of the given enforcer,
```

Comment:
This comment is a bit hard for me to understand. Can we rephrase? Maybe:
```
If a node's label matcher has the same name as a label matcher of the given enforcer, then it will be replaced.
```

Reply:
sounds good :+1: